packages/table/src/Table.tsx

```diff
@@ -0,0 +1,196 @@
+          payload: false,
+        });
+
+        if (boolArray[0]) {
+          dispatch({
+            type: Types.ToggleHeaderCheckedState,
+            payload: true,
+          });
+        } else {
+          dispatch({
+            type: Types.ToggleHeaderCheckedState,
+            payload: false,
+          });
+        }
+      } else {
+        dispatch({
+          type: Types.ToggleHeaderIndeterminate,
+          payload: true,
+        });
+      }
+    }
+  }, [state.data, state.rowCheckedState]);
+
+  let usingHeaderRow = React.useMemo(() => false, [children]);
+  let rows: Array<React.ReactElement>;
+
+  if (typeof children === 'function') {
+    rows = data.map((datum, index) => children({ datum, index }));
+  }
+
+  const sortRows = (columnId: number, key: string) => {
+    dispatch({
+      type: Types.SortTableData,
+      payload: {
+        columnId,
+        key,
+        data,
+      },
+    });
+  };
+
+  const renderHeader = (array: Array<any>): React.ReactNode => {
+    const cols = array.map((child, index) => {
+      if (isComponentType(child, 'HeaderRow')) {
+        usingHeaderRow = true;
+
+        const { children } = child?.props;
+
+        const renderedChildren = Array.isArray(children)
+          ? children
+          : [children];
+
+        return React.cloneElement(child, {
+          children: renderHeader(renderedChildren),
+        });
+      }
+
+      if (isComponentType(child, 'TableHeader')) {
+        const { label, accessor: accessorProp } = child.props;
+
+        let glyph = 'Unsorted';
+
+        const accessor = accessorProp || label?.toLowerCase();
+
+        if (state.sort?.key?.toLowerCase() === accessor) {
+          glyph =
+            state.sort?.direction === 'asc'
+              ? 'SortAscending'
+              : 'SortDescending';
+        }
+
+        return React.cloneElement(child, {
+          // safe until columns need to be reordered
+          key: index,
+          onClick: sortRows,
+          index,
+          glyph,
+        });
+      }
+
+      if (typeof child === 'string') {
+        return (
+          <TableHeader
+            // safe until columns need to be reordered
+            key={index}
+            label={child}
+            index={index}
+            onClick={sortRows}
+          />
+        );
+      }
+
+      return child;
+    });
+
+    return usingHeaderRow ? cols : <HeaderRow>{cols}</HeaderRow>;
+  };
+
+  const renderBody = () => {
+    if (!children) {
+      return null;
+    }
+
+    if (state.selectable) {
+      return rows.map((row, index) => {
+        const selectCell = <CheckboxCell index={index} />;
+
+        return React.cloneElement(row, {
+          children: [selectCell, [...Array.from(row.props.children)]],
+        });
+      });
+    }
+
+    return rows;
+  };
+
+  return (
+    <TableProvider state={state} dispatch={dispatch}>
+      <table cellSpacing="0" cellPadding="0" className={tableStyles} {...rest}>
```

Comment:
We should support `className` to append className styles here.

Reply:
👍